### ISSUE TITLE
onSort event not giving sortAscending ascending property

### ISSUE THREAD

**Issue by jariwalabhavesh** [GriddleGriddle/Griddle#716]
## Expected Behavior
We can also pass custom events in order to controlling paging, sorting etc...
I am facing issue with onSort event, It will return sortProperty with **id** and **sortAscending** property. like
If i am clicking on **name** column then onSort should return:
`{ id: 'name', sortAscending: true }`
and if i again click on **name** column then onSort should return:
`{ id: 'name', sortAscending: false }`

## Actual Behavior
but actually it is returning only **id** property in the both case not the **sortAscending**
`{ id: 'name' }`

## Steps to reproduce
1. Griddle with static data
2. Pass events to griddle

```
 let pagingEvents = {
     onFilter: (filterText) = > {
         console.log("filterText", filterText)
     },
     onSort: (sortProperties) = > {
         console.log("sortProperties", sortProperties)
     },
     onNext: () = > {
         console.log("on Next callback")
     },
     onPrevious: () = > {
         console.log("on Prev callback")
     },
     onGetPage: (pageNumber) = > {
         console.log("pageNumber", pageNumber)
     },
 }

<Griddle data={results} plugins={[plugins.LocalPlugin]} styleConfig={styleConfig} events={pagingEvents}>
    <RowDefinition>
        <ColumnDefinition id="name" title="Name" width={200} style={{ textAlign: 'center' }} />
        <ColumnDefinition id="gender" title="Gender" width={80} style={{ textAlign: 'center' }} />
        <ColumnDefinition id="height" title="Height" width={50} style={{ textAlign: 'center' }} />
    </RowDefinition>
</Griddle>

```

**Comment by jariwalabhavesh** [comment]
@ryanlanciaux @dahlbyk 
I had tried **onSort** event with latest version (1.8.0) and it is working fine when we provide `plugins={[plugins.LocalPlugin]}` with it.

But i am feed data to grid using external api and don't using localplugin because if i externally controlling grid with then paging not working as expected.

But without localplugin, external controlled grid works fine but i am not getting **sortAscending** in onSort Event.()

Thanks.

**Comment by dahlbyk** [comment]
@jariwalabhavesh are you ever calling [`setSortProperties`](https://github.com/GriddleGriddle/Griddle/blob/f57e594f230393717fed579fa97e89effb45242f/src/plugins/local/components/TableHeadingCellContainer.js#L39-L42)?

**Comment by jariwalabhavesh** [comment]
@dahlbyk 
No i am not calling setSortProperties, where can i get method setSortProperties? 
```
 let pagingEvents = {
     onFilter: (filterText) = > {
         console.log("filterText", filterText)
     },
     onSort: (sortProperties) = > {
         console.log("sortProperties", sortProperties)
     },
     onNext: () = > {
         console.log("on Next callback")
     },
     onPrevious: () = > {
         console.log("on Prev callback")
     },
     onGetPage: (pageNumber) = > {
         console.log("pageNumber", pageNumber)
     },
 }

```
I am only getting sortProperties in onSort event, no any function.

**Comment by dahlbyk** [comment]
> where can i get method setSortProperties?

The Griddle module exports [`utils.sortUtils.setSortProperties`](https://github.com/GriddleGriddle/Griddle/blob/992f9c2a22529055dc00e3fb0c5491433eb7e1cb/src/utils/sortUtils.js#L27-L41), which would typically be used with [`actions.setSortColumn`](https://github.com/GriddleGriddle/Griddle/blob/2988f42dd4b3db027587694028783f6507e7848d/src/actions/index.js#L39-L44).

**Comment by jariwalabhavesh** [comment]
@dahlbyk 

i had tried this as you said, Still not receiving **sortAscending**

```
 let pagingEvents = {
            onFilter: (filterText) => { this.setListOptions(filterText, null, 0) },
            onSort: (sortProperties) => {
                let { id, sortAscending} =  sortProperties;
                console.log("sortProperties", sortProperties)
                utils.sortUtils.setSortProperties({setSortColumn: actions.setSortColumn, sortProperty:(sortAscending)?{sortAscending}:null, columnId:id})
            },
            onNext: () => { next() },
            onPrevious: () => { prev() },
            onGetPage: (pageNumber) => { gotoPage(pageNumber - 1) },
        }
```
utils.sortUtils.setSortProperties is calling but no success on it.
I had right or is there any mistake.


I had one confusion also( not able to know what is in griddle state at a perticular time):
Is there any way to see you griddle state in redux-devtools?? should i inject griddle reducer in my project. because if we able to see griddle action and state in redux-devtools then it will be quite issue to see what is actually going on.

**Comment by ibrahima** [comment]
Hmm, I don't know if this is related, but it seems like the sortAscending value I get is the opposite of what I'm seeing in terms of the actual sort. Eg. I'll click on a column and get the ascending sort indicator (verified by using these custom classNames `TableHeadingCellDescending: 'sortIndicator-desc', TableHeadingCellAscending: 'sortIndicator-asc'` and seeing `sortIndicator-asc`) but the sortAscending value I get is false. Then when I click again, it flips to true, but the data is clearly sorted descending, and the indicator class confirms that.

Looking at #717, is it possible that the `onSort` callback is called before `onClick`, and so the sortAscending value is one step out of date? But the weird thing is that the `id` value is always correct, whereas if it was one step behind I'd expect it to say the wrong column name when I switch between sorting different columns.

**Comment by dahlbyk** [comment]
> Looking at #717, is it possible that the `onSort` callback is called before `onClick`, and so the sortAscending value is one step out of date?

You're right, `onSort` being called before `onClick`, means it's called with the old value.

> But the weird thing is that the `id` value is always correct, whereas if it was one step behind I'd expect it to say the wrong column name when I switch between sorting different columns.

Correct. The `sortProperties` value that `onSort` is using is from `sortPropertyByIdSelector`, which finds by `id`. So if you click on a column other than what's currently sorted, `sortProperties` is `null` and `onSort` is called with just the (correct) `id`. Apparently I didn't adequately test #717.

**Comment by ibrahima** [comment]
Ah okay, thanks for confirming that! Luckily I think I can get away without using `sortAscending` for now so it doesn't affect me, but I was trying to test my sorting handler and got very confused by what I was logging.

**Comment by dahlbyk** [comment]
@ibrahima this should be more adequately fixed by https://github.com/GriddleGriddle/Griddle/pull/749.

**Comment by ibrahima** [comment]
Cool, thanks!

**Comment by dahlbyk** [comment]
This has been released with `griddle-react@1.3.0`, though not tagged as `latest` yet.
